src/sbvr-api/permissions.ts

```diff
@@ -920,55 +919,54 @@ const getApiKeyActorIdQuery = _.once(() =>
 );
 const apiActorPermissionError = new PermissionError();
 const getApiKeyActorId = memoize(
-	(apiKey: string) =>
-		getApiKeyActorIdQuery()({
+	async (apiKey: string) => {
+		const apiKeys = (await getApiKeyActorIdQuery()({
 			apiKey,
-		}).then((apiKeys: AnyObject[]) => {
-			if (apiKeys.length === 0) {
-				// We reuse a constant permission error here as it will be cached, and
-				// using a single error instance can drastically reduce the memory used
-				throw apiActorPermissionError;
-			}
-			const apiKeyActorID = apiKeys[0].is_of__actor.__id;
-			if (apiKeyActorID == null) {
-				throw new Error('API key is not linked to a actor?!');
-			}
-			return apiKeyActorID as number;
-		}),
+		})) as AnyObject[];
+		if (apiKeys.length === 0) {
+			// We reuse a constant permission error here as it will be cached, and
+			// using a single error instance can drastically reduce the memory used
+			throw apiActorPermissionError;
+		}
+		const apiKeyActorID = apiKeys[0].is_of__actor.__id;
+		if (apiKeyActorID == null) {
+			throw new Error('API key is not linked to a actor?!');
+		}
+		return apiKeyActorID as number;
+	},
 	{
 		primitive: true,
 		promise: true,
 		maxAge: env.cache.apiKeys.maxAge,
 	},
 );
 
-const checkApiKey = Bluebird.method((req: PermissionReq, apiKey: string) => {
-	if (apiKey == null || req.apiKey != null) {
-		return;
-	}
-	return getApiKeyPermissions(apiKey)
-		.catch(err => {
+const checkApiKey = Bluebird.method(
+	async (req: PermissionReq, apiKey: string) => {
+		if (apiKey == null || req.apiKey != null) {
+			return;
+		}
+		let permissions;
+		try {
+			permissions = await getApiKeyPermissions(apiKey);
+		} catch (err) {
 			console.warn('Error with API key:', err);
 			// Ignore errors getting the api key and just use an empty permissions object.
 			return [];
```

Comment:
I think that this now has to become:
```suggestion
			permissions = [];
```
Otherwise `req.apiKey = {` will no longer be assigned & the `checkApiKey` will no longer be `Bluebird<void>`

Reply:
Yes you're 100% correct, good spot